llvm/utils/TableGen/AIEVariableInstrItineraryEmitter.cpp

```diff
@@ -81,9 +84,18 @@ void AIEVariableInstrItineraryEmitter::emitAltItineraryInfo(raw_ostream &OS) {
     for (const auto &Instr : Instrs)
       OS << "  case " << CurrentNamespace << "::" << Instr << ":\n";
     OS << "  {\n";
-    for (const auto &[RegClass, Itinerary, OpIdx] : RegItineraryPairs) {
-      OS << "    if (checkRCForOperand(" << CurrentNamespace << "::" << RegClass
-         << "RegClass," << OpIdx << "))\n";
+    for (const auto &[Itinerary, RegClassList] : RegItineraryPairs) {
+      assert(RegClassList.empty() == false && "RegClassList cannot be empty");
```

Comment:
nit: `!RegClassList.empty()` ?

Reply:
updating as you suggested, I personally prefer writing it explicitly "`== false`" since a lot of time "`!`" goes unread while going through a code quickly.